src/popup/controller.ts

```diff
@@ -88,11 +88,39 @@ export class InternalControllerStore {
         } as InternalReturnType)
     }
 
-    checkIsConnectedSite = () => {
+    connectedSite = async () => {
+        const oldConnectedSites =
+            await rootStore.walletStore.db.connectedSites.toArray()
+
+        if (
+            oldConnectedSites.some((site) => site.url === this.request?.origin)
+        ) {
+            runInAction(() => {
+                this.connected = true
+            })
+
+            return
+        }
+
+        if (this.request?.origin) {
+            await rootStore.walletStore.db.connectedSites.add({
+                name: this.request?.origin,
+                url: this.request?.origin,
+            })
+
+            const newConnectedSites =
```

Comment:
In my opinion, I think we don't need to differentiate old connectedSites or new connectedSite just connectedSite is fine, but we can leave the comments.

Reply:
I think used the same name is better